internal/models/stepDefinition.go

```diff
@@ -50,15 +50,15 @@ func (sd *StepDefinition) Run(ctx context.Context) (context.Context, interface{}
 		numIn--
 	}
 
-	if len(sd.Args) < numIn {
-		return ctx, fmt.Errorf("%w: expected %d arguments, matched %d from step", ErrUnmatchedStepArgumentNumber, typ.NumIn(), len(sd.Args))
+	if len(s.Args) < numIn {
+		return ctx, fmt.Errorf("%w: expected %d arguments, matched %d from step", ErrUnmatchedStepArgumentNumber, typ.NumIn(), len(s.Args))
 	}
 
 	for i := 0; i < numIn; i++ {
 		param := typ.In(i + ctxOffset)
 		switch param.Kind() {
 		case reflect.Int:
-			s, err := sd.shouldBeString(i)
+			s, err := s.shouldBeString(i)
```

Comment:
This rename introduces slight visual ambiguity by shadowing variable with different type and purpose. In this particular case it is not big deal as new scope (that uses `s` string) if very small and is hard to misinterpret, in larger scopes that could be considered a questionable practice for higher chance of confusion.